### PR TITLE
Sumo Logic output plugin

### PR DESCRIPTION
**Pull request by pmalek-sumo**
This PR adds an output plugin to send data to [Sumo Logic](https://www.sumologic.com/)

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

**Comment by reimda** [comment]
Hi @pmalek-sumo, thanks for the PR.  We have a backlog of new plugins but @ssoroka or I will try to review this soon.

**Comment by pmalek-sumo** [comment]
@reimda Awesome. Thanks!

**Comment by sjwang90** [comment]
@pmalek-sumo If you'd like to submit this as [external plugin](https://github.com/influxdata/telegraf/blob/master/EXTERNAL_PLUGINS.md) in the meantime that would be great.  You and others would be able to start using the plugin with Telegraf sooner without waiting for review.  Here's more information on [external plugins](https://github.com/influxdata/telegraf/blob/master/docs/EXTERNAL_PLUGINS.md).

**Comment by sjwang90** [comment]
Hey @pmalek-sumo.  If you're an employee of Sumo Logic, do you think you and any other employees involved in building things for Influx can sign the [Corporate CLA](https://www.influxdata.com/legal/ccla/)?

File: plugins/outputs/sumologic/sumologic.go [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+	sourceHostHeader     header = `X-Sumo-Host`
+	sourceCategoryHeader header = `X-Sumo-Category`
+	dimensionsHeader     header = `X-Sumo-Dimensions`
+)
+
+type SumoLogic struct {
+	URL               string            `toml:"url"`
+	Timeout           internal.Duration `toml:"timeout"`
+	Method            string            `toml:"method"`
+	MaxRequstBodySize int               `toml:"max_request_body_size"`
```
Comment (ssoroka) [inline plugins/outputs/sumologic/sumologic.go:102]: There is a config.Size type that might be nice for this.

**Review by ssoroka — APPROVED** [review]

File: plugins/outputs/sumologic/sumologic.go (hunk 1 above)
Comment (pmalek-sumo) [reply plugins/outputs/sumologic/sumologic.go]: Oh, I didn't see this. Nice! Changed

**Comment by ssoroka** [comment]
Any idea what's going on with ci macdeps test? it seems like it's running against SumoLogic CI instead of InfluxData?

**Comment by perk-sumo** [comment]
Yeah - all of the CircleCI tests were run against SumoLogic org and not the Telegraf one actually.
We use free plan ATM so we can't build `macdeps`.

**Comment by pmalek-sumo** [comment]
CircleCI project was disconnected from Sumo Logic's repository. I can see the jobs are running now under influx's org.

**Comment by perk-sumo** [comment]
@sjwang90 we've just sent signed CCLA to community@influxdata.com

**Pull request merged**